crates/project_panel/src/project_panel.rs

```diff
@@ -449,30 +455,53 @@ impl ProjectPanel {
     }
 
     fn update_diagnostics(&mut self, cx: &mut ViewContext<Self>) {
-        self.diagnostic_error_paths = self
-            .project
+        let show_warnings =
+            ProjectPanelSettings::get_global(cx).show_diagnostics == ShowDiagnostics::All;
+        self.diagnostics = Default::default();
+        self.project
```

Comment:
is it better here to store everything in a local HashMap first and assign the local variable at the end to `self.diagnostics`?

Reply:
Yeah, I think that's how I'd do it, would make the code nicer, IMHO